### DISCUSSION TITLE
Issue with v4 Endpoints for Content Creation

### DISCUSSION THREAD

**Discussion by rajnishdargan** [Sunbird-Knowlg/Community#221] [General Discussions]
Hi Team,

Following up on the previous discussion [here](https://github.com/orgs/Sunbird-Knowlg/discussions/219), we encountered an issue with the v4 endpoint for content creation.

We were instructed to use the v4 endpoints, but when the generic-editor makes a request like the following to create content:

```
{
  "request": {
    "content": {
      "name": "Untitled Content",
      "code": "1c1d5ed1-7672-487e-8042-d8a97671e59b",
      "mimeType": "application/pdf",
      "createdBy": "ef99949b-7f3a-4a5f-806a-e67e683e38f3",
      "createdFor": ["test-channel"],
      "contentType": "Resource",
      "resourceType": "Learn",
      "creator": "Rahul",
      "framework": "test_k12_framework",
      "organisation": ["Test Channel"],
      "primaryCategory": "Learning Resource"
    }
  }
}

```
We receive the following Bad Request response from the v4 endpoint:

```
{
  "id": "api.content.create",
  "ver": "4.0",
  "ts": "2024-10-03T07:30:28ZZ",
  "params": {
    "resmsgid": "000f26d5-b3a9-4ff7-9092-f3dbcd291a23",
    "msgid": null,
    "err": "VALIDATION_ERROR",
    "status": "failed",
    "errmsg": "contentType cannot be set from request."
  },
  "responseCode": "CLIENT_ERROR",
  "result": {}
}
```

Interestingly, the same request works without any issues when using the v3 endpoint.

It seems the contentType field is causing a validation error in v4, but v3 handles it. We suspect that the generic editors might be outdated or not fully compatible with v4 endpoints. We are thinking of reverting to the v3 endpoints for `/action/content/v3/*` and `/action/content/v1/*` until a resolution is found.

Could someone please guide us to the repository or configuration file where this mapping is defined? 
Is there any middleware or DevOps configuration that handles the endpoint switching?

Thanks!
@pallakartheekreddy @HarishGangula @rajeevsathish @vinukumar-vs @vaibhavbhuva @krgauraw @amitpriyadarshi
@coolbung @itsvick @snehal0904 @arati-tekdi @rushi-tekdi

**Comment by pallakartheekreddy** [comment]
@rajnishdargan Please use `v3` as you are not using Knowlg middleware service.
